addons/slack/template/index.js

```diff
@@ -12,59 +20,56 @@ module.exports = async (ctx) => {
   if (ctx.url.endsWith('events/ingest')) {
     Sdk.debug('Received Events API event');
 
-    const match = ctx.headers.authorization.match(/^bearer\s+(.+)$/i);
-    const token = match && match[1];
-    if (!token || token !== ctx.configuration.fusebit_events_token) {
-      Sdk.debug('Missing or incorrect auth token');
-      return { status: 403 };
-    }
+    if (ctx.caller.permissions) {
+      try {
+        if (ctx.method === 'POST') {
+          let event = ctx.body;
+          Sdk.debug('Valid event', event);
 
-    try {
-      if (ctx.method === 'POST') {
-        let event = ctx.body;
-        Sdk.debug('Valid event', event);
+          // Attach all event handlers off a nested `slack` property
+          let handlers = slack.slack;
+          if (handlers) {
+            const functions = Object.getOwnPropertyNames(handlers)
+              .filter((p) => typeof handlers[p] === 'function')
+              .filter((f) => f == event.type || f == 'all');
+            const client = await getStorageClient(ctx);
+            const promises = functions.map((f) =>
+              handlers[f](sdk, event, {
+                ...ctx.configuration,
+                storage: client,
+              })
+            );
 
-        // Attach all event handlers off a nested `slack` property
-        let handlers = slack.slack;
-        if (handlers) {
-          const functions = Object.getOwnPropertyNames(handlers)
-            .filter((p) => typeof handlers[p] === 'function')
-            .filter((f) => f == event.type || f == 'all');
-          const client = Sdk.getStorageClient(ctx);
-          const promises = functions.map((f) =>
-            handlers[f](sdk, event, {
-              ...ctx.configuration,
-              storage: client,
-            })
-          );
+            Sdk.debug('Invoking handlers', functions);
```

Comment:
Technically, invocation occurred on ln 37; at this point it’s just waiting for completion.

Reply:
Maybe I'm misunderstanding how this work, but I thought on L37 I'm just building the promises, but they aren't executed until I call `Promise.all` on L44?

Reply:
Purely informational: they're all invoked, but execution control doesn't pass to them until the next async operation, whatever it might be (in other words, it's not tied to the Promise.all(), just the next yield).

Reply:
Ok so leave this as-is